src/components/modals/RequestPermissionsModal.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react'
+import { AirshipBridge } from 'react-native-airship'
+
+import s from '../../common/locales/strings'
+import { Checkbox } from '../themed/Checkbox'
+import { MainButton } from '../themed/MainButton'
+import { ModalCloseArrow } from '../themed/ModalParts'
+import { ThemedModal } from '../themed/ThemedModal'
+import { MessageText, TitleText } from '../themed/ThemedText'
+
+interface Props {
+  bridge: AirshipBridge<ChoiceMap>
```

Comment:
Optional: We typically have closing the modal / pressing the X return `undefined`, to indicate that the user desires no changes. Then you can do `if (result == null) return` in the calling code to avoid having any side effects.

This means you could potentially avoid having `enable?: boolean` in the choices map, but I see that you treat the "cancel" button and the modal "X" button as different things, although the all seem to end up in the same place?